src/liballoc/string.rs

```diff
@@ -689,6 +693,7 @@ impl String {
     ///     assert_eq!(String::from("hello"), s);
     /// }
     /// ```
+    #[must_use]
```

Comment:
I think I'd disagree with this one — this is used for FFI when passing back a decomposed `String` to deallocate it.

Reply:
Not in my opinion, no. Idiomatic Rust doesn't tend to use explicit calls to `drop` frequently.

Reply:
I am just a stranger here, but I have some experience with Python, which mantra is "explicit is better than implicit".

Idiomatic Rust doesn't tend to use unsafe blocks frequently, either. I would even say that explicitness is even more valuable in unsafe code.